static/js/typeahead_helper.js

```diff
@@ -126,8 +140,59 @@ exports.compare_by_pms = function (user_a, user_b) {
     return 1;
 };
 
+exports.organize = function(user_a, user_b){
+    var current_subject = $('#subject.recipient_box').val();
+    var current_stream = $('#stream.recipient_box').val();
+    //Filter functions for subject and stream passes the ones that the user itself didn't send and the ones that belong to current subject and stream
+    function pSubject(message){
+        if(message.sent_by_me == false && message.subject == current_subject){
+            return true;
+        } else {
+            return false;    
+        }
+    }
+    function pStream(message){
+        if(message.sent_by_me == false && message.stream == current_stream) { 
+           return true;
+        } else {
+            return false;
+        }
+    }
+    var current_subject_persons_info = message_list.all._items.filter(pSubject);
+    var current_stream_persons_info = message_list.all._items.filter(pStream);
+    var current_subject_persons = [];
+    var current_stream_persons = [];
+    var names_in_current_subject = [];
+    var names_in_current_stream = [];
+    var selected_recipient = "";
+    for(i = 0; i < current_subject_persons_info.length; i += 1){
+         names_in_current_subject.push(current_subject_persons_info[i].sender_full_name);
+    }
+    for(i = 0; i < current_stream_persons_info.length; i += 1){
+        names_in_current_stream.push(current_stream_persons_info[i].sender_full_name);
+    }
+    // Deleting repeated objects in these arrays
+    current_subject_persons = names_in_current_subject.filter(function(elem, pos) {
+        return names_in_current_subject.indexOf(elem) == pos;
+    });
+    if(current_subject_persons.length > 10){
+        current_subject_persons.splice(0, current_subject_persons.length - 10)
+    }
+    current_stream_persons = names_in_current_stream.filter(function(elem, pos) {
+        return names_in_current_stream.indexOf(elem) == pos;
+    });
+    if(current_stream_persons.length > 10){
+        current_stream_persons.splice(0, current_stream_persons.length - 10)
+    }
+    for(i = message_list.all._items.length - 1; i > 0; i -= 1){
+        if($("#zfilt" + String(message_list.all._items[i].id)).hasClass('selected_message') == true ||$("#zhome" + String(message_list.all._items[i].id)).hasClass('selected_message') == true ){
+	    selected_recipient = message_list.all._items[i].sender_full_name;
+        }
+    }
+    return exports.compare_by_pms(user_a, user_b, selected_recipient, current_stream_persons, current_subject_persons); 
+}
 exports.sort_by_pms = function (objs) {
-    objs.sort(exports.compare_by_pms);
```

Comment:
Instead of changing this create a new function like this:
```JavaScript
exports.sort_by_relevance = function (objs) {
    objs.sort(exports.compare_by_relevance);   // 'compare_by_relevance' is your 'organize' function
    return objs;
}
```